File: C/Solution1.c
//Author: Joseph-King
//Date: Jun 22, 2022

#include <stdio.h>

//Returns indices of values in array that's sum equals the target
int* TwoSum(int* arr, int target, const int SIZE, int* test){
    //first loop to go through every index
    for(int i=0; i<SIZE; i++){
        //second loop to go through every index after i
        for(int j=i+1; j<SIZE; j++){
            //determine sum
            int sum = arr[i]+arr[j];

            //if sum = target, print array of indicies
            if(sum == target){
                test[0] = i;
                test[1] = j;

                return(test);
            }
        }
    }

    return(test);
}

//Main function
int main() {
    //size of test cases
    const int SIZE1 = 4;
    const int SIZE2 = 3;
    const int SIZE3 = 2;

    //Array test cases
    int arr1[SIZE1] = {2, 7, 11, 15};
    int arr2[SIZE2] = {3, 2, 4};
    int arr3[SIZE3] = {3, 3};

    //Target test cases
    int target1 = 9;
    int target2 = 6;
    int target3 = 6;

    //Results
    int test[2];

    int* result1 = TwoSum(arr1, target1, SIZE1, test);
    printf("%d %d\n", result1[0], result1[1]);

    int* result2 = TwoSum(arr2, target2, SIZE2, test);
    printf("%d %d\n", result2[0], result2[1]);

    int* result3 = TwoSum(arr3, target3, SIZE3, test);
    printf("%d %d\n", result3[0], result3[1]);

    return 0;
}

File: Java/Solution1.java
//Author: Joseph-King
//Date: Jul 4, 2022

public class Solution1 {
    public static void main(String[] args){
        //Array Test Cases
        int[] arr1 = new int[] {2, 7, 11, 15};
        int[] arr2 = new int[] {3, 2, 4};
        int[] arr3 = new int[] {3, 3};

        //Target Test Cases
        int target1 = 9;
        int target2 = 6;
        int target3 = 6;

        //Run Test Cases
        int[] result1 = twoSum(arr1, target1);
        int[] result2 = twoSum(arr2, target2);
        int[] result3 = twoSum(arr3, target3);

        //Print Test Cases
        printArray(result1);
        printArray(result2);
        printArray(result3);
    }

    //Returns indices of array that equal target
    public static int[] twoSum(int[] arr, int target){
        //Visit every index of array
        for(int i=0; i<arr.length; i++){
            //Visit every index after i
            for(int j=i+1; j<arr.length; j++){
                //Calculate sum
                int sum = arr[i] + arr[j];

                //If sum equals target, return indices in array
                if(sum == target){
                    return(new int[] {i, j});
                }
            }
        }

        //If not found return null
        return(null);
    }

    //Outputs 1D array
    public static void printArray(int[] arr){
        //Visit every index of array
        for(int i=0; i<arr.length; i++){
            //Output using printf, expecting number
            System.out.printf("%d ", arr[i]);
        }
        //End line
        System.out.print("\n");
    }
}